[Rama Venkatesh] iP

ramaven opened the pull request:
# Duke

> "Sometimes our stop-doing list needs to be bigger than our to-do list."
> ~ Patti Digh, Four-Word Self-Help: Simple Wisdom for Complex Lives

Duke frees your mind of having to remember things you need to do. It is:
- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is: 
1. download it from [here](google.com.sg).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features: 
- [x] Managing tasks
- [x] Managing deadlines and events
- [x] Marking item as done
- [ ] Searching for items in your list

If you Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class Main {
        public static void main(String[] args) {
            Application.launch(MainApp.class, args);
        }
 }
```

Commit:
Add Gradle support

ramaven committed:
Level-1

ramaven committed:
Level-2

ramaven committed:
Level-3

ramaven committed:
Level-4

ramaven committed:
Level-5

ramaven committed:
Level-6

ramaven committed:
modified expected output

ramaven committed:
enum & refactoring

ramaven committed:
enum; refactoring

Wilfredwongkc commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,96 @@
+                    System.out.println("   Noted. I've removed this task: ");
+                    System.out.println("  " + tasktoDel.toString());
+                }
+                else if(input.substring(0,8).equals("deadline")){
```
```suggestion
                else if(input.substring(0,8).equals("deadline")) {
```

Wilfredwongkc commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,96 @@
+                    System.out.println("Got it. I've added this task: ");
+                    System.out.print("  " + newTask.toString());
+
+                } else if(input.substring(0,5).equals("event")){
```
```suggestion
                } else if(input.substring(0,5).equals("event")) {
```

Wilfredwongkc commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,96 @@
+                if(input.length()<4 ){
+                    throw new DukeException("Unacceptable input");
+                }
+                if(input.substring(0,4).equals("todo")){
```
```suggestion
                if(input.substring(0,4).equals("todo")) {
```

Wilfredwongkc commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,96 @@
+import tasks.*;
+import exceptions.*;
```
Maybe can name out all the files that you are importing instead of using a wildcard?

Wilfredwongkc commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,96 @@
+                for(int i = 0; i < taskCounter; i++){
+                    System.out.println((i+1) + "." + tasks.get(i).toString());
+                }
+            } else if (input.length() > 4 && input.substring(0,4).equals("done")){
```
```suggestion
            } else if (input.length() > 4 && input.substring(0,4).equals("done")) {
```

Wilfredwongkc commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,96 @@
+        String input = "";
+        try {while(true){
+            input = sc.nextLine();
+            if(input.equals("bye")){
```
```suggestion
            if(input.equals("bye")) {
```

Wilfredwongkc commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,96 @@
+        int taskCounter = 0;
+
+        String input = "";
+        try {while(true){
```
```suggestion
        try {
            while(true) {
```

Wilfredwongkc commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,96 @@
+                break;
+                //System.exit();
+            }
+            else if(input.equals("list")) {
```
```suggestion
            else if (input.equals("list")) {
```

Wilfredwongkc commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,96 @@
+                }
+            } else if (input.length() > 4 && input.substring(0,4).equals("done")){
+                String taskDone = input.substring(5);
+                int taskDoneIndex = Integer.parseInt(taskDone)-1;
```
```suggestion
                int taskDoneIndex = Integer.parseInt(taskDone) - 1;
```

Wilfredwongkc commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,96 @@
+                System.out.println(tasks.get(taskDoneIndex).toString());
+            }
+            else {
+                if(input.length()<4 ){
```
```suggestion
                if (input.length() < 4) {
```

Wilfredwongkc commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,96 @@
+                    taskCounter++;
+                    System.out.println("Got it. I've added this task: ");
+                    System.out.println("  " + newEvent.toString());
+                } else if(input.length()>5 && input.length()<8){
```
```suggestion
                } else if (input.length() > 5 && input.length() < 8) {
```

Wilfredwongkc commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,96 @@
+                } else if(input.length()>5 && input.length()<8){
+                    throw new DukeException("Unacceptable input");
+                }
+                else if(input.substring(0,6).equals("delete")) {
```
```suggestion
                else if (input.substring(0,6).equals("delete")) {
```

Wilfredwongkc commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,96 @@
+
+                System.out.println("\nNow you have " +  (taskCounter) + " tasks in the list.");
+            }
+        }} catch (DukeException e){
+            System.out.println("OOPS!!! You have enteted an invalid category");
+        }
```
I think can format the curly braces better?

Wilfredwongkc reviewed (changes requested):
LGTM! Just some nits on coding standards and then it should be fine.

aakanshanarain commented on src/main/java/Tasks/DeadlineTask.java:
```diff
@@ -0,0 +1,18 @@
+    }
+
+    @Override
+    public String toString(){
```
`public String toString() {`

aakanshanarain commented on src/main/java/Tasks/DeadlineTask.java:
```diff
@@ -0,0 +1,18 @@
+    @Override
+    public String toString(){
+        String typeString = type == TaskType.TODO ? "T" : type == TaskType.EVENT? "E" : "D";
+        String doneSymbol = isDone? "X" : " ";
```
`String doneSymbol = isDone ? "X" : " ";`

aakanshanarain commented on src/main/java/Tasks/EventTask.java:
```diff
@@ -0,0 +1,17 @@
+    @Override
+    public String toString(){
+        String typeString = type == TaskType.TODO ? "T" : type == TaskType.EVENT? "E" : "D";
+        String doneSymbol = isDone? "X" : " ";
```
`String doneSymbol = isDone ? "X" : " ";`

aakanshanarain reviewed (commented):
Looks ok! I think you could work on the formatting of the braces as well as including whitespace in your code. Good luck!

ramaven committed:
added save functionality

ramaven committed:
added save

ramaven committed:
Added DukeDate parse

ramaven committed:
Merge branch 'master' into branch-Level-8

ramaven committed:
Merge pull request #1 from ramaven/branch-Level-8

Added DukeDate parse

ramaven committed:
fixed getDate() bug

ramaven committed:
refactored packages

ramaven committed:
removed debug statements

ramaven committed:
added Command class; OOP

ramaven committed:
adhered coding standard

ramaven committed:
adhered coding standard

ramaven committed:
added javadocs

ramaven committed:
added find functionality

ramaven committed:
Merge pull request #2 from ramaven/branch-A-CodingStandard

Branch a coding standard

ramaven committed:
Merge pull request #3 from ramaven/branch-A-JavaDoc

added javadocs

ramaven committed:
Merge pull request #4 from ramaven/branch-Level-9

added find functionality

ramaven committed:
Merge pull request #6 from ramaven/add-gradle-support

Add Gradle support pull req

ramaven committed:
fixed file save bug

ramaven committed:
fixed file save bug

ramaven committed:
added gradle support

ramaven committed:
Merge branch 'master' into branch-A-Gradle

ramaven committed:
Merge pull request #7 from ramaven/branch-A-Gradle

added gradle support

ramaven committed:
checkstyle

ramaven committed:
Merge pull request #8 from ramaven/branch-A-Checkstyle

checkstyle

ramaven committed:
modified gradle build file

ramaven committed:
Merge branch 'master' into branch-A-Checkstyle

ramaven committed:
Merge pull request #9 from ramaven/branch-A-Checkstyle

modified gradle build file

ramaven committed:
Added GUI

ramaven committed:
Merge pull request #10 from ramaven/branch-Level-10

Added GUI

ramaven committed:
Add assert statements

No assert statements are present.

Add statements are needed to declare for
things that ought to be true at that point in time

ramaven committed:
Merge pull request #11 from ramaven/branch-A-Assertions

Add assert statements

ramaven committed:
Find command: make matching case insensitive

Find command is case sensitive

A case insensitive find is more user-friendly because users cannot be expected to remember the exact case of the keywords

Let's,
* update the search algorithm to use case-insensitive matching

ramaven committed:
Merge pull request #12 from ramaven/branch-C-BetterSearch

Find command: make matching case insensitive

ramaven committed:
Improve code quality

Coding standard not fully adhered to and SLAP principle not followed in Ui class

Adhering more closely to coding standards makes code more readable and better
understood, and following coding principles makes it easier for other programmers to
understand it.

Let's
* fix the order of import statements
* follow SLAP principle in Ui class' init method
* simplify complicated expressions in FindCommand class

ramaven committed:
Merge pull request #13 from ramaven/branch-A-CodeQuality

Improve code quality

ramaven committed:
Add product screenshot

ramaven committed:
Add user guide

ramaven committed:
Set theme jekyll-theme-minimal

ramaven committed:
Set theme jekyll-theme-leap-day

ramaven committed:
Edit User Guide

ramaven committed:
Merge branch 'master' of https://github.com/ramaven/ip

ramaven committed:
Update README

ramaven committed:
Update README.md

ramaven committed:
Update README.md

ramaven committed:
Update README.md

ramaven committed:
Add credits

ramaven committed:
Improve code quality

ramaven committed:
Add test methods
